metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -186,20 +177,22 @@ private void publish(RdfWrapper rdf, Date recordDate, List<String> datasetIdsToR
     final FullBeanImpl fullBean = fullBeanConverter.convertRdfToFullBean(rdf);
 
     // Provide the preprocessor: this will set the created and updated timestamps as needed.
-    final TriConsumer<FullBeanImpl, FullBeanImpl, Date> fullBeanPreprocessor =
+    final TriConsumer<FullBeanImpl, FullBeanImpl, Pair<Date, Date>> fullBeanPreprocessor =
         preserveUpdateAndCreateTimesFromRdf ? EMPTY_PREPROCESSOR
             : (FullBeanPublisher::setUpdateAndCreateTime);
 
-    //The list can be empty if redirects are not applied if no redirect were found
-    List<String> recordsForRedirection = checkAndApplyRedirects(rdf, recordDate,
-        datasetIdsToRedirectFrom, performRedirects);
+    //The list can be empty if redirects are not applied or if no redirect were found
+    List<Pair<String, Date>> recordsForRedirection = RecordRedirectsUtil
+        .checkAndApplyRedirects(recordRedirectDao::createUpdate,
+            recordRedirectDao::getRecordRedirectByNewId, rdf, recordDate, datasetIdsToRedirectFrom,
+            performRedirects, this::getSolrDocuments);
```

Comment:
Why not pass recordRedirectDao as a parameter either to the method or on construction of a RecordRedirectsUtil instance?

Reply:
Functionally I wanted transparency on the utils class, but I have changed it to provide a recordRedirectDao instead.